File: datn_fe/src/app/service/statistics/statistics.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class StatisticsService {
  private apiUrl = `${environment.apiUrl}/api/statistics`;

  constructor(private http: HttpClient) { }

  /**
   * Get revenue statistics
   */
  getRevenueStatistics(): Observable<any> {
    return this.http.get(`${this.apiUrl}/revenue`);
  }

  /**
   * Get top combo purchases
   */
  getTopCombos(limit: number = 10): Observable<any> {
    return this.http.get(`${this.apiUrl}/top-combos`, { params: { limit: limit.toString() } });
  }

  /**
   * Get top subject purchases
   */
  getTopSubjects(limit: number = 10): Observable<any> {
    return this.http.get(`${this.apiUrl}/top-subjects`, { params: { limit: limit.toString() } });
  }

  /**
   * Get all statistics summary
   */
  getStatisticsSummary(): Observable<any> {
    return this.http.get(`${this.apiUrl}/summary`);
  }
}

File: datn_be/src/main/java/com/dntn/datn_be/service/StatisticsService.java
package com.dntn.datn_be.service;

import com.dntn.datn_be.dto.common.ResponseGlobalDto;
import java.util.Map;

public interface StatisticsService {
    ResponseGlobalDto<Map<String, Object>> getRevenueStatistics();
    ResponseGlobalDto<Object> getTopCombos(Integer limit);
    ResponseGlobalDto<Object> getTopSubjects(Integer limit);
    ResponseGlobalDto<Map<String, Object>> getStatisticsSummary();
}

